src/apps/api/views/datasets.py

```diff
@@ -63,10 +62,48 @@ def create(self, request, *args, **kwargs):
 
     def destroy(self, request, *args, **kwargs):
         # TODO: Confirm this has a test
-        if request.user != self.get_object().created_by:
-            raise PermissionDenied()
+        instance = self.get_object()
+
+        error = self.check_delete_permissions(request, instance)
+
+        if error:
+            return Response(
+                {'error': error},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
         return super().destroy(request, *args, **kwargs)
 
+    @action(detail=False, methods=('POST',))
+    def delete_many(self, request):
+        qs = Data.objects.filter(id__in=request.data)
+        errors = {}
+
+        for dataset in qs:
+            error = self.check_delete_permissions(request, dataset)
+            if error:
+                errors[dataset.name] = error
+
+        if not errors:
+            qs.delete()
+
+        return Response(
+            errors if errors else {'detail': 'Datasets deleted successfully'},
+            status=status.HTTP_400_BAD_REQUEST if errors else status.HTTP_200_OK
+        )
+
+    def check_delete_permissions(self, request, dataset):
+        if request.user != dataset.created_by:
+            return 'Cannot delete a dataset that is not yours'
```

Comment:
I think we can simplify this a bit by raising a `ValidationError(msg)` or some such? `PermissionDenied(msg)` maybe?

Reply:
I figured it should be done this way so we can return an object with every failure. I believe if we raise an error, it won't continue through the rest of the loop, and we only get one failure. Is that an okay way to do this?

Reply:
Ah, right, good point, makes sense. Maybe add a comment about this?